libshpool/src/daemon/server.rs

```diff
@@ -1010,8 +1035,18 @@ impl Server {
         Ok(env)
     }
 
+    // Generate the path to the env file that is populated on every
+    // attach.
+    fn session_env_file(&self, session_name: PathBuf) -> PathBuf {
+        self.session_dir(session_name).join("forward.env")
+    }
+
     fn ssh_auth_sock_symlink(&self, session_name: PathBuf) -> PathBuf {
-        self.runtime_dir.join("sessions").join(session_name).join("ssh-auth-sock.socket")
+        self.session_dir(session_name).join("ssh-auth-sock.socket")
+    }
+
+    fn session_dir(&self, session_name: PathBuf) -> PathBuf {
```

Comment:
nit: session_name can be a Path, to avoid unnecessary clones

Reply:
Good call. I wound up having to do `<P: AsRef<Path>>(... : P)` rather than using Path directly since Path is a trait.